tests/test_sql/test_flush_utils.py

```diff
@@ -1,47 +1,146 @@
+    current_django_version = django.VERSION
+    django.VERSION = version
+    yield version
+    django.VERSION = current_django_version
+
+
+class TestFlushDjangoMigrationTable(object):
     """Ensure ``connection.ops`` methods are called with expected args."""
-    style = Style()
-    testing_connection_mock = mocker.MagicMock()
-    testing_connection_mock.introspection.get_sequences.return_value = []
-    connections_mock = mocker.patch('django.db.connections._connections')
-    connections_mock.test = testing_connection_mock
-    sql.flush_django_migrations_table('test', style)
-    testing_connection_mock.ops.sql_flush.assert_called_once_with(
-        style,
-        [sql.DJANGO_MIGRATIONS_TABLE_NAME],
-        [],
-        allow_cascade=False,
+
+    _style = Style()
+
+    @pytest.mark.skipif(
+        django.VERSION < (3, 1),
+        reason='requires `Django>=3.1`',
     )
-    testing_connection_mock.ops.execute_sql_flush.assert_called_once_with(
-        'test',
-        mocker.ANY,
+    def test_django31_calls(self, mocker):
+        """Check that the calls are right on Django 3.1."""
+        testing_connection_mock = self._mock_connection(mocker)
+        sql.flush_django_migrations_table('test', self._style)
+        testing_connection_mock.ops.sql_flush.assert_called_once_with(
+            self._style,
+            [sql.DJANGO_MIGRATIONS_TABLE_NAME],
+            reset_sequences=True,
+            allow_cascade=False,
+        )
+        testing_connection_mock.ops.execute_sql_flush.assert_called_once_with(
+            mocker.ANY,
+        )
+
+    @pytest.mark.skipif(
+        django.VERSION < (2, 0) or django.VERSION >= (3, 1),
+        reason='requires `2.0<=Django<3.1`',
     )
+    def test_django20_calls(self, mocker):
+        """Check that the calls are right on Django 2.0 - 3.0."""
+        testing_connection_mock = self._mock_connection(mocker)
+        sql.flush_django_migrations_table('test', self._style)
+        testing_connection_mock.ops.sql_flush.assert_called_once_with(
+            self._style,
+            [sql.DJANGO_MIGRATIONS_TABLE_NAME],
+            sequences=[],
+            allow_cascade=False,
+        )
+        testing_connection_mock.ops.execute_sql_flush.assert_called_once_with(
+            mocker.ANY,
+            mocker.ANY,
+        )
+
+    @pytest.mark.skipif(
+        django.VERSION >= (2, 0),
+        reason='requires `Django<2.0`',
+    )
+    def test_django1_11_calls(self, mocker):
+        """Check that the calls are right on Django < 2.0."""
+        testing_connection_mock = self._mock_connection(mocker)
+        sql.flush_django_migrations_table('test', self._style)
+        testing_connection_mock.ops.sql_flush.assert_called_once_with(
+            self._style,
+            [sql.DJANGO_MIGRATIONS_TABLE_NAME],
+            sequences=[],
+            allow_cascade=False,
+        )
+
+    def _mock_connection(self, mocker):
```

Comment:
NIP: you can even create fixture returning mocked connection

Reply:
Good point, that would be cleaner, I'll just do that.